rero_ils/modules/locations/api.py

```diff
@@ -69,20 +70,24 @@ def extended_validation(self, **kwargs):
         return True
 
     @classmethod
-    def get_pickup_location_pids(cls, patron_pid=None):
+    def get_pickup_location_pids(cls, patron_pid=None, item_barcode=None):
         """Return pickup locations."""
         from ..patrons.api import Patron
-        from ..patron_types.api import PatronType
-        search = LocationsSearch()\
-            .filter('term', is_pickup=True)
+        from ..items.api import Item
+        search = LocationsSearch()
+
+        if item_barcode:
+            item = Item.get_item_by_barcode(item_barcode)
+            item_loc = item.get_location()
+            queries = [Q('term', pid=p) for p in item_loc.restrict_pickup_to]
+            search = search.query(Q('bool', should=queries))
```

Comment:
as I understood this can be replaced by a terms filter: https://elasticsearch-dsl.readthedocs.io/en/latest/search_dsl.html#filters

Reply:
Sometimes you search hours and hours and at often at the end @jma found a simple solution :-)